traffic_stats/influxdb_tools/create/create_ts_databases.go

```diff
@@ -20,45 +20,81 @@ under the License.
 package main
 
 import (
-	"flag"
 	"fmt"
-	"os"
 
 	influx "github.com/influxdata/influxdb/client/v2"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
 )
 
-func main() {
+const (
+	cache           = "cache_stats"
+	deliveryService = "deliveryservice_stats"
+	daily           = "daily_stats"
+)
+
+func createFlags() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			Name:  "url",
+			Usage: "The influxdb url and port",
+			Value: "http://localhost:8086",
+		},
+		cli.IntFlag{
+			Name:  "replication",
+			Usage: "The number of nodes in the cluster",
+			Value: 3,
+		},
+		cli.StringFlag{
+			Name:  "user",
+			Usage: "The influxdb username used to create DBs",
+			Value: "",
+		},
+		cli.StringFlag{
+			Name:  "password",
+			Usage: "The influxdb password used to create DBs",
+			Value: "",
+		},
+	}
+}
+
+func create(c *cli.Context) error {
+	influxURL := c.String("url")
+	replication := c.Int("replication")
+	user := c.String("user")
+	password := c.String("password")
 
-	influxURL := flag.String("url", "http://localhost:8086", "The influxdb url and port")
-	replication := flag.String("replication", "3", "The number of nodes in the cluster")
-	user := flag.String("user", "", "The influxdb username used to create DBs")
-	password := flag.String("password", "", "The influxdb password used to create DBs")
-	flag.Parse()
-	fmt.Printf("creating datbases for influxUrl: %v with a replication of %v using user %s\n", *influxURL, *replication, *user)
+	fmt.Printf("creating datbases for influxUrl: %s with a replication of %d using user %s\n", influxURL, replication, user)
 	client, err := influx.NewHTTPClient(influx.HTTPConfig{
-		Addr:     *influxURL,
-		Username: *user,
-		Password: *password,
+		Addr:     influxURL,
+		Username: user,
+		Password: password,
 	})
 	if err != nil {
-		fmt.Printf("Error creating influx client: %v\n", err)
-		os.Exit(1)
+		return errors.Wrap(err, "Error creating influx client")
 	}
 	_, _, err = client.Ping(10)
 	if err != nil {
-		fmt.Printf("Error creating influx client: %v\n", err)
-		os.Exit(1)
+		return errors.Wrap(err, "Error creating influx client")
 	}
 
 	createCacheStats(client, replication)
 	createDailyStats(client, replication)
 	createDeliveryServiceStats(client, replication)
+	return nil
 }
 
-func queryDB(client influx.Client, cmd string) (res []influx.Result, err error) {
+// queryDB takes a variadic argument for the target database so as to make
+// passing the variable optional, however, if passed, only the first db passed
+// in will be used
+func queryDB(client influx.Client, cmd string, dbs ...string) (res []influx.Result, err error) {
+	db := ""
+	if len(dbs) > 0 {
+		db = dbs[0]
```

Comment:
I'm all for reuse of code. There aren't any API contracts we need to adhere to. I'd prefer to just fix these outright.